internal/dag/gatewayapi_processor.go

```diff
@@ -269,32 +285,34 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 
 	for _, rule := range route.Spec.Rules {
 
-		matchconditions := []*matchConditions{}
+		var matchconditions []*matchConditions
 
 		for _, match := range rule.Matches {
 			mc := &matchConditions{}
-			// TODO: Replace this with nil check when gateway-api includes this fix -
-			// https://github.com/kubernetes-sigs/gateway-api/commit/9d63656df8cc9e67da60d4fe3f6289aad22e72d3
-			if match.Path.Value == "" || match.Path.Type == "" {
-				match.Path = gatewayapi_v1alpha1.HTTPPathMatch{Type: gatewayapi_v1alpha1.PathMatchPrefix, Value: "/"}
+			if match.Path == nil || match.Path.Value == nil || match.Path.Type == nil {
+				match.Path = &gatewayapi_v1alpha1.HTTPPathMatch{Type: pathMatchTypePtr(gatewayapi_v1alpha1.PathMatchPrefix), Value: pointer.StringPtr("/")}
 			}
-			switch match.Path.Type {
-			case gatewayapi_v1alpha1.PathMatchPrefix:
-				mc.pathMatchConditions = append(mc.pathMatchConditions, &PrefixMatchCondition{Prefix: stringOrDefault(match.Path.Value, "/")})
-			case gatewayapi_v1alpha1.PathMatchExact:
-				mc.pathMatchConditions = append(mc.pathMatchConditions, &ExactMatchCondition{Path: stringOrDefault(match.Path.Value, "/")})
-			default:
-				routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonPathMatchType, "HTTPRoute.Spec.Rules.PathMatch: Only Prefix match type and Exact match type are supported.")
+			if match.Path.Type != nil {
+				switch *match.Path.Type {
+				case gatewayapi_v1alpha1.PathMatchPrefix:
+					mc.pathMatchConditions = append(mc.pathMatchConditions, &PrefixMatchCondition{Prefix: stringOrDefaultFromPtr(match.Path.Value, "/")})
+				case gatewayapi_v1alpha1.PathMatchExact:
+					mc.pathMatchConditions = append(mc.pathMatchConditions, &ExactMatchCondition{Path: stringOrDefaultFromPtr(match.Path.Value, "/")})
+				default:
+					routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonPathMatchType, "HTTPRoute.Spec.Rules.PathMatch: Only Prefix match type and Exact match type are supported.")
+				}
 			}
 
 			if match.Headers != nil {
-				switch match.Headers.Type {
-				case gatewayapi_v1alpha1.HeaderMatchExact:
-					for k, v := range match.Headers.Values {
-						mc.headerMatchCondition = append(mc.headerMatchCondition, HeaderMatchCondition{MatchType: HeaderMatchTypeExact, Name: k, Value: v})
+				if match.Headers.Type != nil {
```

Comment:
ah maybe this was missing before, what do we do if header match type is not supplied? should we default to something?

Reply:
If there's no header match then there isn't a default, only if nothing is defined do we get the default prefix match on "/".